base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -3830,29 +3830,23 @@ where
         {
             (
                 match tx_type {
-                    TxType::PaymentToOther | TxType::Burn => TransactionDirection::Outbound,
-                    TxType::PaymentToSelf |
+                    TxType::PaymentToOther |
+                    TxType::Burn |
+                    TxType::CodeTemplateRegistration |
+                    TxType::ValidatorNodeRegistration |
                     TxType::CoinSplit |
```

Comment:
`TxType::PaymentToSelf`, `TxType::CoinSplit` and `TxType::CoinJoin` must all be treated similarly, in this case `TransactionDirection::Outbound`

Reply:
For that matter, all change direction metadata except `TxType::Coinbase` should be treated similarly
With `TxType::Coinbase` there can be no change